geo-traits/src/line_string.rs

```diff
@@ -0,0 +1,76 @@
+use crate::{Dimension, LineStringIterator, PointTrait};
+use geo_types::{Coord, CoordNum, LineString};
+
+/// A trait for accessing data from a generic LineString.
+pub trait LineStringTrait: Sized {
+    /// The coordinate type of this geometry
+    type T: CoordNum;
+
+    /// The type of each underlying coordinate, which implements [PointTrait]
+    type ItemType<'a>: 'a + PointTrait<T = Self::T>
+    where
+        Self: 'a;
+
+    /// The dimension of this geometry
+    fn dim(&self) -> Dimension;
+
+    /// An iterator over the points in this LineString
+    fn points(&self) -> LineStringIterator<'_, Self::T, Self::ItemType<'_>, Self> {
```

Comment:
Oh that sounds great. It also means that in theory an implementor could create their own iterator, that doesn't have to be `LineStringIterator`? 

> Since we're iterating over points, I'd expect this to be called something like `PointIterator`

I think of this as "iterating over a LineString", not "iterating over points"

Reply:
I think my usage is more conventional. The stdlib is typically a great place to look for idioms. Unfortunately for this case in particular, most of the stdlib stuff is very generic, so it's a little hard to draw corollaries:

Some examples:

> [`Vec::windows(&self, size: usize) -> Windows<'_, T>`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.windows): Returns an iterator over all contiguous windows of length size. 

It's an iterator over the *Windows* not over the *Vec*.

> [`Strings::chars`](https://doc.rust-lang.org/std/primitive.str.html#method.chars): Returns an iterator over the [char](https://doc.rust-lang.org/std/primitive.char.html)s of a string slice.

It's an iterator over the *chars* not over the *string slice*.

I might be wrong, but I'd like you to show some compelling prior art for this nomenclature if we're going to merge it.

Alternatively, since it's now returning a generic impl Trait, we could just make them private and then I don't care very much what they're called.

Reply:
> Alternatively, since it's now returning a generic `impl Trait`, we could just make them private and then I don't care very much what they're called.

Oh brilliant! I didn't realize they no longer needed to be public, and I'm very happy with them being private! I made them private in [`7a2effd` (#1157)](https://github.com/georust/geo/pull/1157/commits/7a2effd48f581b52b0ae450bc1be8e85754b468b).